core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2035,6 +2039,13 @@ protected Scope visitAliasedRelation(AliasedRelation relation, Optional<Scope> s
 
             RelationType descriptor = relationType.withAlias(relation.getAlias().getValue(), aliases);
 
+            boolean canProcessHiddenColumn = aliases == null;
```

Comment:
This is not ideal. It's relying on an internal non-obvious detail of `withAlias`.